File: 20_Buzzword/javascript/buzzword.js
// BUZZWORD
//
// Converted from BASIC to Javascript by Oscar Toledo G. (nanochess)
//

function print(str)
{
    document.getElementById("output").appendChild(document.createTextNode(str));
}

function input()
{
    var input_element;
    var input_str;
    
    return new Promise(function (resolve) {
                       input_element = document.createElement("INPUT");
                       
                       print("? ");
                       input_element.setAttribute("type", "text");
                       input_element.setAttribute("length", "50");
                       document.getElementById("output").appendChild(input_element);
                       input_element.focus();
                       input_str = undefined;
                       input_element.addEventListener("keydown", function (event) {
                                                      if (event.keyCode == 13) {
                                                      input_str = input_element.value;
                                                      document.getElementById("output").removeChild(input_element);
                                                      print(input_str);
                                                      print("\n");
                                                      resolve(input_str);
                                                      }
                                                      });
                       });
}

function tab(space)
{
    var str = "";
    while (space-- > 0)
        str += " ";
    return str;
}

var a = ["",
         "ABILITY","BASAL","BEHAVIORAL","CHILD-CENTERED",
         "DIFFERENTIATED","DISCOVERY","FLEXIBLE","HETEROGENEOUS",
         "HOMOGENEOUS","MANIPULATIVE","MODULAR","TAVISTOCK",
         "INDIVIDUALIZED","LEARNING","EVALUATIVE","OBJECTIVE",
         "COGNITIVE","ENRICHMENT","SCHEDULING","HUMANISTIC",
         "INTEGRATED","NON-GRADED","TRAINING","VERTICAL AGE",
         "MOTIVATIONAL","CREATIVE","GROUPING","MODIFICATION",
         "ACCOUNTABILITY","PROCESS","CORE CURRICULUM","ALGORITHM",
         "PERFORMANCE","REINFORCEMENT","OPEN CLASSROOM","RESOURCE",
         "STRUCTURE","FACILITY","ENVIRONMENT",
         ];

// Main program
async function main()
{
    print(tab(26) + "BUZZWORD GENERATOR\n");
    print(tab(15) + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n");
    print("\n");
    print("\n");
    print("\n");
    print("THIS PROGRAM PRINTS HIGHLY ACCEPTABLE PHRASES IN\n");
    print("'EDUCATOR-SPEAK' THAT YOU CAN WORK INTO REPORTS\n");
    print("AND SPEECHES.  WHENEVER A QUESTION MARK IS PRINTED,\n");
    print("TYPE A 'Y' FOR ANOTHER PHRASE OR 'N' TO QUIT.\n");
    print("\n");
    print("\n");
    print("HERE'S THE FIRST PHRASE:\n");
    do {
        print(a[Math.floor(Math.random() * 13 + 1)] + " ");
        print(a[Math.floor(Math.random() * 13 + 14)] + " ");
        print(a[Math.floor(Math.random() * 13 + 27)] + "\n");
        print("\n");
        y = await input();
    } while (y == "Y") ;
    print("COME BACK WHEN YOU NEED HELP WITH ANOTHER REPORT!\n");
}

main();

File: 20_Buzzword/lua/buzzword.lua
--- Buzzword
--- Ported by Brian Wilkins.
--- Updated for modern buzzwords and corporate-speak

print [[
                        BUZZWORD GENERATOR
              CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY

]]

print [[


THIS PROGRAM PRINTS HIGHLY ACCEPTABLE PHRASES IN
'EDUCATOR-SPEAK' THAT YOU CAN WORK INTO REPORTS
AND SPEECHES.  WHENEVER A QUESTION MARK IS PRINTED,
TYPE A 'Y' FOR ANOTHER PHRASE OR 'N' TO QUIT.
]]

local phraseList = {"ABILITY","BASAL","BEHAVIORAL","CHILD-CENTERED",
               "DIFFERENTIATED","DISCOVERY","FLEXIBLE","HETEROGENEOUS",
               "HOMOGENEOUS","MANIPULATIVE","MODULAR","TAVISTOCK",
               "INDIVIDUALIZED","LEARNING","EVALUATIVE","OBJECTIVE",
               "COGNITIVE","ENRICHMENT","SCHEDULING","HUMANISTIC",
               "INTEGRATED","NON-GRADED","TRAINING","VERTICAL AGE",
               "MOTIVATIONAL","CREATIVE","GROUPING","MODIFICATION",
               "ACCOUNTABILITY","PROCESS","CORE CURRICULUM","ALGORITHM",
               "PERFORMANCE","REINFORCEMENT","OPEN CLASSROOM","RESOURCE",
               "STRUCTURE","FACILITY","ENVIRONMENT"}

--- Credit to https://stackoverflow.com/a/33468353/19232282
--- for the pickPhrase function
local copyPhraseList = {}

function pickPhrase()
   local i
   -- make a copy of the original table if we ran out of phrases
   if #copyPhraseList == 0 then
      for k,v in pairs(phraseList) do
         copyPhraseList[k] = v
      end
   end
               
   -- pick a random element from the copy  
   i = math.random(#copyPhraseList)
   phrase = copyPhraseList[i] 
               
   -- remove phrase from copy
   table.remove(copyPhraseList, i)
               
   return phrase
end

--- Reused from Bagels.lua
function getInput(prompt)
    io.write(prompt)
    io.flush()
    local input = io.read("l") 
    if not input then  --- test for EOF
        print("COME BACK WHEN YOU NEED HELP WITH ANOTHER REPORT!")
        os.exit(0)
    end
    return input
end

for i=1,3,1 do
    ::phrasepick::
    io.write (pickPhrase() .. " ")
    io.write (pickPhrase() .. " ")
    io.write (pickPhrase())
    print()
    io.stdin:flush()
    local response = getInput("? ")
    if response:match("[yY].*") then
       goto phrasepick
    else
       print("COME BACK WHEN YOU NEED HELP WITH ANOTHER REPORT!")
       os.exit(0)
    end
end

